hypershift-operator/controllers/nodepool/aws.go

```diff
@@ -69,6 +81,13 @@ func awsMachineTemplateSpec(infraName, ami string, hostedCluster *hyperv1.Hosted
 		tags[tag.Key] = tag.Value
 	}
 
+	// We enforce the AWS cluster cloud provider tag here.
+	// Otherwise, this would race with the HC defaulting itself hostedCluster.Spec.Platform.AWS.ResourceTags.
+	key := awsClusterCloudProviderTagKey(hostedCluster.Spec.InfraID)
+	if _, ok := tags[key]; !ok {
+		tags[key] = infraLifecycleOwned
```

Comment:
by the ut failure, it looks like you can get here and `tags` == nil

Reply:
thanks, updated!